onnxruntime/core/graph/graph_utils.cc

```diff
@@ -104,7 +207,7 @@ bool RemoveSingleInSingleOutNode(Graph& graph, Node& node) {
   graph.RemoveNode(node.Index());
 
   // Add new edge connecting the input with the output nodes directly.
-  graph.AddEdge(input_edge_node, output_edge_node,
```

Comment:
My main concern is whether the above new action for the implicit inputs should be included in the AddEdge method. Currently, AddEdge also checks and updates the InputDefFor ex of the output_edge_node to be the same as the OutputDef of the input_edge_node.
I am saying this because each time we call an AddEdge, we will need to do the above.
For example, it can be added inside the `if (src_arg != dst_arg)` check of AddEdge(). What do you think?

Reply:
I'm not sure this is required in the majority of AddEdge scenarios. The implicit input update is only required when re-routing an output after removing a node that was providing an implicit input to a subgraph. I can see the subgraph logic being applicable to all 'remove node/s' scenarios, but not many AddEdge scenarios.

Reply:
Yes, probably it is too much. I just remembered that we do the update of the dest node arg to be the same as the input one (which is also happening only after edge removal). But maybe we don't want to complicate the AddEdge method too much, at least not for now.